HttpSimulator/SimulatedHttpRequest.cs

```diff
@@ -154,8 +168,13 @@ public override string GetServerVariable(string name)
             switch (name) {
                 case "HTTP_USER_AGENT":
                     return GetKnownRequestHeader(HttpWorkerRequest.HeaderUserAgent);
-            }   
-            return base.GetServerVariable(name);
+            }
+			var value = _serverVariables.Get(name);
+			if(value != null)
+			{
+				return value;
+            }
```

Comment:
hey can you reformat this method.

Reply:
Ok. 
I will also add unit test. 
Found one issue.